File: frontend/src/types/team.ts
export type TeamInput = {
  team_name: string;
  accepting_members: boolean;
};

export type TeamMemberResponse = {
  user_id: string;
  username: string;
  is_leader: boolean;
  joined_at: string;
};

export type TeamResponse = {
  id: number;
  team_name: string;
  invite_code: string;
  accepting_members: boolean;
  members: Array<TeamMemberResponse>;
} | null;

export type TeamMember = {
  id: string;
  userName: string;
  isLeader: boolean;
  joinedAt: string;
};

export type Team = {
  id: number;
  teamName: string;
  inviteCode: string;
  acceptingMembers: boolean;
  members: Array<TeamMember>;
};

File: backend/app/db/mappers/team.py
from app.db.models import Team
from app.schemas.team import TeamOut, TeamMemberOut


def team_to_out(team: Team) -> TeamOut:
    return TeamOut(
        id=team.id,
        team_name=team.team_name,
        invite_code=team.invite_code,
        accepting_members=team.accepting_members,
        members=[
            TeamMemberOut(
                user_id=m.user_id,
                username=m.user.username,
                is_leader=m.is_leader,
                joined_at=m.joined_at,
            )
            for m in team.members
        ],
    )
